src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -361,6 +366,21 @@ private void BuildLinkTag(TagHelperAttributeList attributes, TagHelperContent bu
             builder.Append("/>");
         }
 
+        private object AppendFileVersionIfApplicable(object attributeValue)
```

Comment:
This refactoring turns things means fallback handling is inconsistent and results in duplicate checks of `AppendVersion` in `BuildLinkTag()`. Also doesn't seem related to fixing #2382. Feel free to follow up later with a separate cleanup PR if you feel strongly about the small amount of duplicate code.